l1-contracts/contracts/bridge/ntv/NativeTokenVault.sol

```diff
@@ -416,28 +428,20 @@ abstract contract NativeTokenVault is INativeTokenVault, IAssetHandler, Ownable2
     /// @param _erc20Data The ERC20 metadata of the token deployed.
     /// @return The address of the beacon proxy (bridged token).
     function _deployBridgedToken(
-        uint256 _originChainId,
+        uint256 _tokenOriginChainId,
         bytes32 _assetId,
         address _originToken,
         bytes memory _erc20Data
     ) internal returns (address) {
-        bytes32 salt = _getCreate2Salt(_originChainId, _originToken);
-
-        BeaconProxy l2Token = _deployBeaconProxy(salt);
-        uint256 tokenOriginChainId = BridgedStandardERC20(address(l2Token)).bridgeInitialize(
-            _assetId,
-            _originToken,
-            _erc20Data
-        );
-        // an extra check for legacy tokens on L1, they might not be registered i.e.
-        if (block.chainid == L1_CHAIN_ID && tokenOriginChainId == 0) {
-            revert L1TokenDeploymentWithZeroChainId(_assetId);
-        }
-        tokenOriginChainId = tokenOriginChainId == 0 ? L1_CHAIN_ID : tokenOriginChainId;
-        if (tokenOriginChainId == block.chainid) {
+        bytes32 salt = _getCreate2Salt(_tokenOriginChainId, _originToken);
+
+        BeaconProxy l2Token = _deployBeaconProxy(salt, _tokenOriginChainId);
+        BridgedStandardERC20(address(l2Token)).bridgeInitialize(_assetId, _originToken, _erc20Data);
+        if (_tokenOriginChainId == block.chainid) {
```

Comment:
we can do this check at the top of the function for failing faster/wasting less gas